common/FileSystem.cpp

```diff
@@ -271,6 +274,134 @@ std::wstring FileSystem::GetWin32Path(std::string_view str)
 	return ret;
 }
 
+bool FileSystem::CreateWin32IShellLink(const std::string_view path_suffix, const std::string_view launch_arguments,
+	const std::string_view description, const GUID& base_directory_ID)
```

Comment:
If you use a pointer here, you can both check the pointer for null instead of the thing it points to, which makes a lot more sense, and also you can forward declare GUID instead of including the entire windows header into `Filesystem.h`.